StyleCop.Analyzers/StyleCop.Analyzers/SpacingRules/SA1018CodeFixProvider.cs

```diff
@@ -52,10 +53,16 @@ public override FixAllProvider GetFixAllProvider()
                 if (!previousToken.HasTrailingTrivia)
                     continue;
 
-                SyntaxToken corrected = previousToken.WithoutTrailingWhitespace().WithoutFormatting();
-                Document updatedDocument = context.Document.WithSyntaxRoot(root.ReplaceToken(previousToken, corrected));
-                context.RegisterCodeFix(CodeAction.Create("Remove space", t => Task.FromResult(updatedDocument)), diagnostic);
+                context.RegisterCodeFix(CodeAction.Create("Remove space", t => GetTransformedDocument(context, root, previousToken)), diagnostic);
```

Comment:
:exclamation: Move as much of the logic from this loop to the callback as possible.